### PR TITLE
Add Heketi framework to gluster block provisioner code

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: gluster/block/glusterblock-provisioner.go [hunk 1]
```diff
@@ -161,25 +177,64 @@ func (p *glusterBlockProvisioner) Provision(options controller.VolumeOptions) (*
...
+		if err != nil {
+			glog.Errorf("glusterfs: error creating volume %v ", err)
+			return nil, fmt.Errorf("error creating volume %v", err)
+		}
+
+		p.volConfig.IQN = "iqn.2016-12.org.gluster-block:aafea465-9167-4880-b37c-2c36db8562ea"
+		p.volConfig.TargetPortal = "192.168.1.11"
+
+	default:
+		return nil, fmt.Errorf("error parsing value for 'opmode' for volume plugin %s", provisionerName)
```
Comment (wongma7) [inline gluster/block/glusterblock-provisioner.go]: you could parse it in parseClassParameters first just for consistency

File: gluster/block/glusterblock-provisioner.go [hunk 2]
```diff
@@ -161,25 +177,64 @@ func (p *glusterBlockProvisioner) Provision(options controller.VolumeOptions) (*
...
 	}
-	return dtarget, diqn, nil
+	return &p.volConfig, nil
+
+	/*
```
Comment (wongma7) [inline gluster/block/glusterblock-provisioner.go]: delete this commented block? or does gofmt need to be run on this file, something about it looks off about it!

File: gluster/block/glusterblock-provisioner.go (hunk 1 above)
Comment (humblec) [inline gluster/block/glusterblock-provisioner.go]: Will address it in the following PR. @wongma7

File: gluster/block/glusterblock-provisioner.go (hunk 2 above)
Comment (humblec) [inline gluster/block/glusterblock-provisioner.go]: Done.

**Comment by wongma7** [comment]
One of heketi's dependencies is missing so it doesn't build:
```
../../vendor/github.com/heketi/heketi/client/api/go-client/client.go:21:2: cannot find package "github.com/dgrijalva/jwt-go" in any of:
```

Also somehow this package got removed:
```
controller/controller_test.go:36:2: cannot find package "k8s.io/client-go/kubernetes/fake" in any of:
```
After adding dgrijalva/jwt-go to glide.yaml can you try re-running "glide up -v" then "glide-vc --use-lock-file".

There has to be an easier way to add new dependency packages, let me find out for you tomorrow...

**Comment by humblec** [comment]
@wongma7 Done.!

**Comment by wongma7** [comment]
lgtm, CI build is working 👍

**Pull request merged**